### PR TITLE
COMMERCE-9459 | Shipments - Link to carrier based tracking

### PR DESCRIPTION
**Pull request by iovaneale**
DEV: https://issues.liferay.com/browse/COMMERCE-9459

QA: https://issues.liferay.com/browse/COMMERCE-9671

### CODE REVIEW

**Comment by iovaneale** [comment]
ci:test:relevant

**Comment by iovaneale** [comment]
ci:test:commerce-functional

**Comment by iovaneale** [comment]
ci:stop

**Comment by iovaneale** [comment]
ci:test:relevant

**Comment by iovaneale** [comment]
ci:test:commerce-functional

**Comment by iovaneale** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceChannels.testcase [hunk 1]
```diff
@@ -527,6 +527,94 @@ definition {
 			locator1 = "CommerceEntry#ENTRY_NAME_BODY_VERTICAL_ELLIPSIS");
 	}
 
+	@description = "This is a test for COMMERCE-9459. CanAddEditAndRemoveTrackingURLToShippingMethod"
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceChannels.testcase:530]: Can you rename it to `CanEditShippingMethodTrackingURL`?

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceShipments.testcase [hunk 2]
```diff
@@ -8,11 +8,17 @@ definition {
 	property testray.main.component.name = "Order Management";
 
 	setUp {
-		CommerceConfiguration.commerceSetUp();
+		User.firstLoginPG();
+
+		//CommerceConfiguration.commerceSetUp();
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceShipments.testcase:13]: Why do we have comments here?

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceShipments.testcase [hunk 3]
```diff
@@ -438,6 +642,109 @@ definition {
 			locator1 = "WorkflowMetrics#SEARCH_FIELD_MAPPINGS_METRICS_INDEXES");
 	}
 
+	@description = "This is a test for COMMERCE-9459. CanAddCarrierTrackingURLFromShipment"
+	@priority = "3"
+	test CanAddCarrierTrackingURLFromShipment {
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceShipments.testcase:647]: Rename it to CanAddShipmentTrackingURL

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceShipments.testcase [hunk 4]
```diff
@@ -593,6 +900,129 @@ definition {
 		}
 	}
 
+	@description = "This is a test for COMMERCE-9459. CanEditCarrierTrackingURLFromShipment"
+	@priority = "3"
+	test CanEditCarrierTrackingURLFromShipment {
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceShipments.testcase:905]: Rename it to CanEditShipmentTrackingURL

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceShipments.testcase [hunk 5]
```diff
@@ -2666,4 +3096,113 @@ definition {
 			value1 = "7.00");
 	}
 
+	@description = "This is a test for COMMERCE-9459. ViewShipmentCarrierTrackingURL"
+	@priority = "3"
+	test ViewShipmentCarrierTrackingURL {
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceShipments.testcase:3101]: Rename it to CanViewShipmentTrackingURL

**Review by alessiorendina — CHANGES_REQUESTED** [review]
@iovaneale please check my comments, thanks!

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceShipments.testcase (hunk 5 above)
Comment (iovaneale) [reply modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceShipments.testcase:3101]: Ok

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceChannels.testcase (hunk 1 above)
Comment (iovaneale) [reply modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceChannels.testcase:530]: Well, I could, yet I believe that just "Edit" doesn't reflect the actual test.

**Comment by iovaneale** [comment]
ci:stop

**Comment by iovaneale** [comment]
ci:test:relevant

**Comment by iovaneale** [comment]
ci:test:sf

**Comment by iovaneale** [comment]
ci:test:commerce-functional

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceChannels.testcase [hunk 6]
```diff
@@ -527,6 +527,94 @@ definition {
...
+	@priority = "2"
+	test CanAddEditAndRemoveTrackingURLToShippingMethod {
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceChannels.testcase]: Rename it to CanEditShippingMethodTrackingURL

Comment (alessiorendina) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceChannels.testcase]: Rename it to CanEditShippingMethodTrackingURL

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceShipments.testcase [hunk 7]
```diff
@@ -8,11 +8,17 @@ definition {
...
+
+		//CommerceConfiguration.commerceSetUp();
+
 	}
 
 	tearDown {
-		CommerceConfiguration.commerceTearDown();
+		User.logoutPG();
+
+		//CommerceConfiguration.commerceTearDown();
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceShipments.testcase]: Please remove comments

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceShipments.testcase [hunk 8]
```diff
@@ -331,6 +337,204 @@ definition {
 			locator1 = "CommerceOrders#ORDER_WAREHOUSE_BY_ACTIVE_STATUS");
 	}
 
+	@description = "This is a test for COMMERCE-9459"
+	@priority = "3"
+	test AssertShipmentTrackingURLChangesWithShippingOption {
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceShipments.testcase:336]: Rename it to CanCheckShipmentTrackingURLAfterShippingMethodSelection

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceShipments.testcase (hunk 3 above)
Comment (alessiorendina) [reply modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceShipments.testcase]: This is not resolved yet, please rename it

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceShipments.testcase (hunk 4 above)
Comment (alessiorendina) [reply modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceShipments.testcase]: This is not resolved yet, please rename it

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceShipments.testcase (hunk 5 above)
Comment (alessiorendina) [reply modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceShipments.testcase]: This is not resolved yet, please rename it

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceShipments.testcase [hunk 9]
```diff
@@ -3098,7 +3094,7 @@ definition {
 
 	@description = "This is a test for COMMERCE-9459. ViewShipmentCarrierTrackingURL"
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceShipments.testcase]: ViewShipmentCarrierTrackingURL -> CanViewShipmentTrackingURL

**Review by alessiorendina — CHANGES_REQUESTED** [review]
@iovaneale Thank you for your work, please check my comments, I'll do the changes in another PR. In general, when we fix a commit we do a fixup, it helps the review.

**Comment by alessiorendina** [comment]
Forwarded here: #2889 

@iovaneale you can close this one, thanks.

**Pull request closed without merging**